src/nix/flake.cc

```diff
@@ -658,12 +658,16 @@ struct CmdFlakeCheck : FlakeCommand
 
                         else if (
                             name == "lib"
+                            || name == "configurations"
                             || name == "darwinConfigurations"
                             || name == "darwinModules"
+                            || name == "debug"
                             || name == "flakeModule"
                             || name == "flakeModules"
                             || name == "herculesCI"
                             || name == "homeConfigurations"
+                            || name == "internals"
+                            || name == "modules"
                             || name == "nixopsConfigurations"
```

Comment:
Hold on guys, are we now adding not only nixpkgs specific code to nix but also random flakes specific code? :-1:

Reply:
`configurations` and `modules` are for #6257. I don't suppose those are a problem?

Do you mean `debug` and `internals`? I think ignoring those is useful for any flake that wants to be easy to troubleshoot.

These changes are quite intentional and not random or specific to any flake?

If you mean that we shouldn't add code specific to the flakes feature, I can agree with that. It would be nice to separate that out, but I don't think that's a particularly productive choice of component to spin off. I'd rather do that with the dev shell logic, a component whose independent development would pay off more, and where it would make sense to have pinnable or alternate implementations.

Reply:
Sorry, I meant the old ones.
`configurations` and `modules` are fine but don't solve anything
instead of `internals` and `debug` maybe it would be nice to keep a single attribute since both of them generally are going to be used for the same thing

Reply:
I like the separation. For instance, `internals` could be used in a shebang script, whereas `debug` is only for human consumption and therefore can be assumed that messing around with it is ok.
Using `debug` from a shebang script seems like a bad idea, but debugging using `internals` isn't quite as bad. I suppose `internals.debug` could be a convention, but the indirection is unnecessary and a bit annoying when you're using it in the repl. So `internals` is definitely the one to keep, but `debug` is also worth the tiny cost.